app/experimenter/nimbus-ui/src/components/PageResults/GraphsWeekly/index.tsx

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import React, { useState } from "react";
+import Collapse from "react-bootstrap/Collapse";
+import vegaEmbed, { VisualizationSpec } from "vega-embed";
+import {
+  BRANCH_COMPARISON,
+  BRANCH_COMPARISON_TITLE,
+} from "../../../lib/visualization/constants";
+import { lineGraphConfig } from "../../../lib/visualization/graphConfig";
+import {
+  AnalysisDataWeekly,
+  BranchDescription,
+  FormattedAnalysisPoint,
+} from "../../../lib/visualization/types";
+import { ReactComponent as CollapseMinus } from "./minus.svg";
+import { ReactComponent as ExpandPlus } from "./plus.svg";
+
+/**
+ * We should only graph if there are at least 2 weeks of data
+ * because otherwise the graph is just a single point and useless.
+ *
+ * So we return empty mergedBranchData to avoid graphing when there
+ * isn't enough data
+ */
+const getMergedBranchData = (
+  probeSetSlug: string,
+  weeklyResults: { [branch: string]: BranchDescription },
+) => {
+  const mergedBranchData: { [graphID: string]: FormattedAnalysisPoint[] } = {};
+  let maxWeeks = 0;
+  Object.keys(weeklyResults).forEach((branch: string) => {
+    if (probeSetSlug in weeklyResults[branch].branch_data) {
+      Object.values(BRANCH_COMPARISON).forEach((branchComparison) => {
+        const graphID = `${probeSetSlug}_${branchComparison}_graph`;
+        const branchData =
+          weeklyResults[branch].branch_data[probeSetSlug][branchComparison].all;
+        branchData.forEach((dataPoint: FormattedAnalysisPoint) => {
+          dataPoint["branch"] = branch;
+          const weekIndex: number =
+            "window_index" in dataPoint ? dataPoint["window_index"]! : 0;
+          if (weekIndex > maxWeeks) {
+            maxWeeks = weekIndex;
+          }
+        });
+        const updatedBranchData = mergedBranchData[graphID] || [];
+        mergedBranchData[graphID] = updatedBranchData.concat(branchData);
+      });
+    }
+  });
+
+  return maxWeeks === 1 ? {} : mergedBranchData;
+};
+
+const embedGraphs = (
+  probeSetSlug: string,
+  probeSetName: string,
+  mergedBranchData: { [graphID: string]: FormattedAnalysisPoint[] },
+) => {
+  let branchComparisonKey: keyof typeof BRANCH_COMPARISON;
+  for (branchComparisonKey in BRANCH_COMPARISON) {
+    const branchComparison = BRANCH_COMPARISON[branchComparisonKey];
+    const graphID = `${probeSetSlug}_${branchComparison}_graph`;
```

Comment:
This gets constructed a couple times, I wonder if it's possible to make it a little string template and then substitute in the parameters, just so if we ever need to change it it's in one place?